src/dataecon/I.jl

```diff
@@ -185,31 +231,49 @@ function _make_axis(de::DEFile, rng::AbstractVector{<:Union{Symbol,AbstractStrin
     return ax_id[]
 end
 
+# _get_axis_of(de::DEFile, vec::NTuple, dim=1) = _make_axis(de, Base.axes1(vec))
 _get_axis_of(de::DEFile, vec::AbstractVector, dim=1) = _make_axis(de, Base.axes1(vec))
 _get_axis_of(de::DEFile, vec::AbstractUnitRange, dim=1) = _make_axis(de, vec)
 _get_axis_of(de::DEFile, vec::AbstractArray, dim=1) = _make_axis(de, Base.axes(vec, dim))
 
+function _get_axis_range_firstdate(axis::C.axis_t)
+    @assert axis.ax_type == C.axis_range
+    FR = _to_julia_frequency(Val(axis.frequency))
+    return _unpack_date(FR, axis.frequency, axis.first)
+end
+
 #############################################################################
 # write tseries and mvtseries
 
+
+_de_array_data(; eltype, elfreq=C.freq_none, obj_type, nbytes, val) = (; eltype, elfreq, obj_type, nbytes, val)
```

Comment:
:-)  this is not my best moment :-)

The logic here is that all methods of `_to_de_array` would return a sturcture with all relevant fields, which are then used in `_do_store_array`. But instead of defining an actual `struct`, I am using a `NamedTuple` (sort of like an anonymous struct, if you will). This particular function, `_de_array_data`, acts as "constructor" - so you call it with keyword arguments and it constructs the `NamedTuple` for you. 

This was easier to work with while prototyping, because Revise.jl doesn't handle changes in a strcuct - you have to restart the repl.